### ISSUE TITLE
Problem with Print Ordered Sheet Index_Revit 2016- Sequel

### ISSUE THREAD

**Issue by gmb99** [eirannejad/pyRevit#368]
Hi Ehsan

I see you closed out our previous thread in which I think we resolved the corruption and ordering issues of sheet names. 

I would like to resolve the issue of getting a sheet set printed which corresponds to the sheets listed in a schedule. I have attached an image of the sheet schedule which lists a grand total of 126 sheets to be printed. I have set up 2 Filters which restrict the schedule to certain sheets which have a sheet number that contains N23 and a Sheet Category which contains 90&. The sheet schedule correctly lists those sheets and does not contain other sheets which do not meet those filters.

The PyRevit Tools list those sheets as 125 sheets which may makes sense since it starts with Zero. But when I print that set to PDF it prints 147 sheets. The good news is that it has printed the 126 sheets in the correct order but it has also printed an additional 21 sheets that have been added to the set. Those 21 additional sheets are checked in the Revit Print Range dialog box but they shouldn't be printed if PyRevit is controlling the printing from its own interface. I have had similar problems in the past with the Revit dialog box limiting the number of sheets printed from PyRevit according to which sheets are checked in the Revit Dialog box. For Instance PyRevit printed only 12 sheets from the same sheet schedule containing 126 sheets because the Revit Print range only had 12 sheets checked. I shouldn't have to go to the Revit Print Range and manually update this number to correspond to the number of sheets listed in the Sheet schedule.

What is the interaction between the Revit Print Range Dialog and the PyRevit interface? If the Revit Print dialog controls the number of sheets printed as seems to be the case then it seems that PyRevit is not updating the Revit Print range before it prints.

Let me know what more information you need

And thank again for your help

Greg

![20180811_1343_pyrevit_print sheets from schedule_126_125](https://user-images.githubusercontent.com/41219070/43996198-6370f898-9d72-11e8-811c-2a92600da737.png)
![20180811_1343_pyrevit_print sheets from schedule_126_125_bluebeam prints 147 sheets](https://user-images.githubusercontent.com/41219070/43996199-6389e3a8-9d72-11e8-9fe8-cb20810e4279.png)
Greg

**Comment by eirannejad** [comment]
Oh okay. I didn’t know this wasn’t resolved. pyRevit print tool overwrites the printset it creates for the print job. So you shouldn’t ever need to make changes to the Revit print window other than the printer defaults. 

Can I run tests on the old model or you’ll send me a new one?

**Comment by gmb99** [comment]
HI Ehsan

I uploaded a file to the drop box folder that I have shared with you. File is dated with todays date.

**Comment by eirannejad** [comment]
Okay guys,
just pushed an update that fixes this issue. Please test and close if resolved.

**Comment by gmb99** [comment]
Hi Ehsan

Not sure what the Red Malignant Bug sign means

Good news: I was able to get an ordered print set from the file that we have tested together. Right number of sheets, in the right order etc. Congratulations

The not as good news is that the same procedure is not working on my sister project which has the same issues of sheets not correctly ordering themselves in the schedule or the browser. A shift click to get the sheets ordered is generating an error log. See attached.

I feel like we are getting closer and I would love the good news to spread to any Revit file with a sheet schedule that we apply the PyRevit print command to.

Thanks as always for your help

All the best Greg

![20180817 pyrevit version](https://user-images.githubusercontent.com/41219070/44294523-2fa00d80-a24d-11e8-8541-3a158c6157e5.png)
![20180817_drawing index schedule](https://user-images.githubusercontent.com/41219070/44294524-3038a400-a24d-11e8-9b30-d11107be4ad2.png)
![20180817_l200_145_pyrevit error log_on shift print](https://user-images.githubusercontent.com/41219070/44294525-3038a400-a24d-11e8-97da-a1fb6300d116.png)

**Comment by gmb99** [comment]
Hi Ehsan

OK yes. You called it. That project I am told, inserted Unicode characters into the sheet numbers to allow the creation of duplicate sheet numbers for the purpose of providing a permit set of drawings. So yes that must be what is creating the errors.
This office has struggled with the best way to carry multiple sheet sets within a project for the purpose of permit set or other documentation that must be produced while a project is in progress. Obviously the problem with inserting Unicode is nobody knows it is there and it creates problems like that which we have encountered. I will try to set up a detached copy of this file and remove duplicate sheets and try to run again.
Any thoughts about how to easily find sheets with hidden Unicode in them?

Thanks

Greg

**Comment by eirannejad** [comment]
I think I can make a tool for that. Maybe to list all sheets with special characters in the sheet names?

**Comment by gmb99** [comment]
Yes that would be very helpful to identify where the special characters have been inserted. If you were not the person to place the special characters in the sheet numbers you would not know they were there and it is hard to find them.

**Comment by eirannejad** [comment]
Ya I agree. Okay I'll add that. So should I close this issue?

**Comment by gmb99** [comment]
Ehsan

I exported to excel the sheet names and was able to track down the duplicate sheets that were causing  the PyRevit errors. Once the duplicate sheets were removed , (which must have had hidden Unicode in the sheet numbers), I was able to shift click the PyRevit print tool to re-order the sheets correctly and to print an ordered set with the correct number of sheets. So congratulations - I believe we have sorted this issue out and you can close the issue. 

Thanks again

Greg

**Comment by eirannejad** [comment]
Hey Greg,
I added an option to the `Spy > List Elements` tool to list `Sheets with Hidden Characters` in the sheet number. It looks for unicode non-printable characters in sheet numbers and will list those. Hope this helps.

**Comment by gmb99** [comment]
Hi Ehsan

Thanks for the follow up. I tried it out and it gave back reams of information that was overwhelming. You probably understand what it all means, but I am not sure how to interpret it or use it. Is there that much Unicode in the sheet names?
Thanks
Greg

![20180822_pyrevit_list elements_sheet with unicode](https://user-images.githubusercontent.com/41219070/44502695-948da600-a647-11e8-9077-fef2f9f0e2e3.png)

**Comment by eirannejad** [comment]
Holy moly!!! Okay so in front of each sheet numbers there is a `unicode:` section and a `bytes:` section. Disregard the `\x**` in the bytes section for now.
But each `\u200e` in the unicode section, represents a non-printable Left-to-Right Unicode mark. Which means that there are close to 180 non-printable characters before the sheet numbers!!
